fe/fe-core/src/main/java/org/apache/doris/planner/GroupCommitPlanner.java

```diff
@@ -0,0 +1,181 @@
+import org.apache.thrift.TSerializer;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+// Used to generate a plan fragment for a group commit
+// we only support OlapTable now.
+public class GroupCommitPlanner {
+    private static final Logger LOG = LogManager.getLogger(GroupCommitPlanner.class);
+
+    private Database db;
+    private OlapTable table;
+    private TUniqueId loadId;
+    private Backend backend;
+    private TExecPlanFragmentParamsList paramsList;
+    private ByteString execPlanFragmentParamsBytes;
+
+    public GroupCommitPlanner(Database db, OlapTable table, List<String> targetColumnNames, TUniqueId queryId)
+            throws UserException, TException {
+        this.db = db;
+        this.table = table;
+        TStreamLoadPutRequest streamLoadPutRequest = new TStreamLoadPutRequest();
+        if (targetColumnNames != null) {
+            streamLoadPutRequest.setColumns(String.join(",", targetColumnNames));
+        }
+        streamLoadPutRequest
+                .setDb(db.getFullName())
+                .setMaxFilterRatio(1)
+                .setTbl(table.getName())
+                .setFileType(TFileType.FILE_STREAM).setFormatType(TFileFormatType.FORMAT_CSV_PLAIN)
+                .setMergeType(TMergeType.APPEND).setThriftRpcTimeoutMs(5000).setLoadId(queryId)
+                .setGroupCommit(true);
+        StreamLoadTask streamLoadTask = StreamLoadTask.fromTStreamLoadPutRequest(streamLoadPutRequest);
+        StreamLoadPlanner planner = new StreamLoadPlanner(db, table, streamLoadTask);
+        // Will using load id as query id in fragment
+        TExecPlanFragmentParams tRequest = planner.plan(streamLoadTask.getId());
+        for (Map.Entry<Integer, List<TScanRangeParams>> entry : tRequest.params.per_node_scan_ranges.entrySet()) {
+            for (TScanRangeParams scanRangeParams : entry.getValue()) {
+                scanRangeParams.scan_range.ext_scan_range.file_scan_range.params.setFormatType(
+                        TFileFormatType.FORMAT_PROTO);
+                scanRangeParams.scan_range.ext_scan_range.file_scan_range.params.setCompressType(
+                        TFileCompressType.PLAIN);
+            }
+        }
+        List<TScanRangeParams> scanRangeParams = tRequest.params.per_node_scan_ranges.values().stream()
+                .flatMap(Collection::stream).collect(Collectors.toList());
+        Preconditions.checkState(scanRangeParams.size() == 1);
+        loadId = queryId;
+        // see BackendServiceProxy#execPlanFragmentsAsync
+        paramsList = new TExecPlanFragmentParamsList();
+        paramsList.addToParamsList(tRequest);
+        execPlanFragmentParamsBytes = ByteString.copyFrom(new TSerializer().serialize(paramsList));
+    }
+
+    public Future<PGroupCommitInsertResponse> prepareGroupCommitInsertRequest(ConnectContext ctx,
```

Comment:
`public PGroupCommitInsertResponse executeGroupCommitInsert(ConnectContext ctx, List<InternalService.PDataRow> rows)` is better?